x/gov/handler.go

```diff
@@ -214,6 +241,36 @@ func (h TallyHandler) validate(ctx weave.Context, db weave.KVStore, tx weave.Tx)
 	return &msg, proposal, nil
 }
 
+// TallyResultExecutor functionality to be executed on successful proposal.
+type TallyResultExecutor func(db weave.KVStore, p *Proposal) error
+
+// noOpExecutor return nil without doing anything else
+func noOpExecutor(_ weave.KVStore, _ *Proposal) error {
+	return nil
+}
+
+// updateElectorateExecutor contains logic to update an electorate.
+func updateElectorateExecutor(elecBucket *ElectorateBucket) TallyResultExecutor {
+	return func(db weave.KVStore, proposal *Proposal) error {
+		details := proposal.GetElectorateUpdateDetails()
```

Comment:
This doesn't check permissions? Any passing vote can update any electorate?

Reply:
No, this is executed only when a tally result is `accepted`. See https://github.com/iov-one/weave/pull/581/files#diff-c51941a9f6c50cb3936120ad31fa8b98R200 . All the logic for tally result calculation is still in place and happens before. IMHO there are no further constraints required.

Reply:
No, this is executed only for an accepted tally. See L193. All the logic for accepting/ rejecting a proposal is still in place and happened before

Reply:
Okay, assume two elections rules are in the system.

1. 2/3 majority, electorate governance board
2. 1/2 majortiy, you me and Piotr as members

If I understand the code correctly (please make a test case to prove me wrong), then I could make a proposal to update the governance board using election rules #2. Then you and me vote on it and change the governance board.

That is, while there are restrictions on updating the electorate (to admin), they don't seem to be enforced if this is due to a tally. I would keep the permission check and just let eg. an election rule be the admin of an electorate.

Reply:
The governance board would be configured with election rule 1 for example (https://github.com/iov-one/weave/pull/581/files#diff-ac7a84d658879fb883e8697ad86373daR19) 

Then on `CreateElectorateUpdateProposalMsg` it is set from the electorate
https://github.com/iov-one/weave/pull/581/files#diff-c51941a9f6c50cb3936120ad31fa8b98R387

The missing part is the genesis initializer that connects the electorate with its update rule. Thought I had added it but it is not impl. 🙈 Will do now!

Reply:
Okay, this is fine to merge now. But brings up a good point to think about.